lib/transaction.py

```diff
@@ -737,7 +737,6 @@ def sign(self, keypairs):
                     sig = private_key.sign_digest_deterministic(pre_hash, hashfunc=hashlib.sha256, sigencode = ecdsa.util.sigencode_der)
                     assert public_key.verify_digest(sig, pre_hash, sigdecode = ecdsa.util.sigdecode_der)
                     txin['signatures'][j] = bh2u(sig) + int_to_hex(self.nHashType() & 255, 1)
-                    txin['x_pubkeys'][j] = pubkey
```

Comment:
I still don't know what negative effects this could have on code that expects this 'buggy' behavior or otherwise idiosyncratic behavior downstream of this -- have you done due diligence on that?  :)

Reply:
I tried some partial signing and serialise/deserialise.

I looked over the code and now I'm unsure if the `get_sorted_pubkeys()` can  change the order of the keys.  Maybe it is better to explicitly assign the x_pubkey instead.